pike/path.py

```diff
@@ -0,0 +1,690 @@
+"""
+Path-like interface for working with a Tree object
+"""
+from __future__ import absolute_import
+
+import datetime
+import io
+import os
+import sys
+import warnings
+
+try:
+    from pathlib import PureWindowsPath, _WindowsFlavour
+except ImportError:
+    from pathlib2 import PureWindowsPath, _WindowsFlavour
+
+from . import model
+from . import ntstatus
+from . import nttime
+from . import smb2
+
+
+MAX_SYMLINK_RECURSE = 10
+
+NOT_FOUND_STATUSES = (
+    ntstatus.STATUS_OBJECT_NAME_NOT_FOUND,
+    ntstatus.STATUS_OBJECT_PATH_NOT_FOUND,
+)
+
+
+class _PikeFlavour(_WindowsFlavour):
+    """
+    Implement pike-specific overrides of WindowsFlavour semantics.
+
+    Handle the fact that ``_drv`` may be a :py:class:`pike.model.Tree` instance.
+    """
+
+    def casefold(self, s):
```

Comment:
Looks legit!!! The case fold I suppose is to be sync w/ windows case insensitivity.